meilisearch-http/src/routes/indexes/search.rs

```diff
@@ -81,14 +79,42 @@ impl From<SearchQueryGet> for SearchQuery {
     }
 }
 
+// TODO: TAMO: split on :asc, and :desc, instead of doing some weird things
+
+/// Transform the sort query parameter into something that matches the post expected format.
+fn fix_sort_query_parameters(sort_query: &str) -> Vec<String> {
+    let mut sort_parameters = Vec::new();
+    let mut merge = false;
+    for current_sort in sort_query.trim_matches('"').split(',').map(|s| s.trim()) {
+        if current_sort.starts_with("_geoPoint(") {
+            sort_parameters.push(current_sort.to_string());
+            merge = true;
+        } else if merge && !sort_parameters.is_empty() {
+            sort_parameters
+                .last_mut()
+                .unwrap()
+                .push_str(&format!(",{}", current_sort));
+            if current_sort.ends_with("):desc") || current_sort.ends_with("):asc") {
+                merge = false;
+            }
+        } else {
+            sort_parameters.push(current_sort.to_string());
+            merge = false;
+        }
+    }
+    sort_parameters
+}
+
```

Comment:
Since I can run the tests now I'll rewrite this function after my review :smile: